staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -912,25 +911,20 @@ func ValidateCustomResourceDefinitionOpenAPISchema(schema *apiextensions.JSONSch
 	}
 
 	if len(schema.XValidations) > 0 {
-		if schemaIsNonStructural {
-			allErrs = append(allErrs, field.Forbidden(fldPath.Child("x-kubernetes-validations"), "x-kubernetes-validations is only supported in structural schema"))
+		structural, err := structuralschema.NewStructural(schema)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("x-kubernetes-validations"), schema.XValidations, "x-kubernetes-validations is only supported in structural schema: "+err.Error()))
 		} else {
-			structural, err := structuralschema.NewStructural(schema)
-			// should not have error since schemaIsNonStructural checked
-			if err != nil {
-				allErrs = append(allErrs, field.Forbidden(fldPath.Child("x-kubernetes-validations"), "x-kubernetes-validations is only supported in structural schema: "+err.Error()))
+			compResults := cel.Compile(structural)
+			if compResults.Error != nil {
```

Comment:
Is this actually an internal error or user error?

Reply:
Updated the error as following:
1. XValitions specified with non-structural schema: `Invalid`
2. schema failed to convert to structural(ideally should not happen): `Invalid`
3. compilation setup failure(e.g. cel env initialize failure): `InternalError`
4. Rule is not specified: `Required`
5. Compilation failed for one rule: `Invalid`
Please let me know if anything needs to be changed. Thank you 🙏